repository/src/main/java/com/epam/esm/repository/api/TagRepository.java

```diff
@@ -0,0 +1,13 @@
+package com.epam.esm.repository.api;
+
+import com.epam.esm.entities.Tag;
+
+import java.util.Optional;
+
+/**
+ * Repository for order entity. May perform {@link Savable}, {@link Deletable}, {@link Countable} and
+ * {@link Findable} operations.
+ */
+public interface TagRepository extends Findable<Tag>, Deletable<Tag>, Savable<Tag>, Countable {
```

Comment:
Could you explain the purpose of all these interfaces?

Reply:
By the requirements different operations on enttites must be performed: Save, get and delete on tag, only get on user and so on. So I segregated entity interfaces on smaller interfaces for more flexible control.